File: Data Structures & Algorithms/house-robber/submission-0.cpp
class Solution {
public:
    int countWays(vector<int>& nums, vector<int>& dp, int i){
        if(i>=nums.size())return 0;
        if (dp[i]!=-1) return dp[i];
        return dp[i]=max(nums[i]+countWays(nums,dp,i+2),countWays(nums,dp,i+1));
    }
    int rob(vector<int>& nums) {
        vector<int> dp(nums.size(),-1);
        return countWays(nums,dp,0);
    }
};

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {boolean}
     */
    containsNearbyDuplicate(nums, k) {
        let freq=new Map();
        for(let i=0;i<nums.length;i++){
            if(freq.has(nums[i]))
            freq.get(nums[i]).push(i);
            else freq.set(nums[i],[i]);
        }
        for(let v of freq.values()){
            for(let i=1;i<v.length;i++){
                if(Math.abs(v[i]-v[i-1])<=k)return true;
            }
        }
        return false;
    }
}
